tests/ui/issues/issue-66667-function-cmp-cycle.stderr

```diff
@@ -15,41 +15,65 @@ LL |     second == 1
    = note: expected fn item `fn() {second}`
                  found type `{integer}`
 
+error[E0308]: mismatched types
```

Comment:
Due to the changes to typeck, we end up emitting this additional error. I can work around it pretty easily by checking if the output type of the binop references errors before returning `builtin_return_ty` a few files up, but I also think that it's fine? Like, this is a meaningful error :sweat_smile: